src/gtk/dc.cpp

```diff
@@ -237,6 +240,76 @@ void* wxGTKCairoDCImpl::GetCairoContext() const
     return cr;
 }
 
+bool wxGTKCairoDCImpl::TryGetWindowSize(wxSize& size, wxSize& sizeMM) const
+{
+    if ( !m_window )
+        return false;
+
+    GdkWindow* const w = gtk_widget_get_window(m_window->GetHandle());
+    if ( !w )
+        return false;
+
+#if GTK_CHECK_VERSION(3, 22, 0)
+    if ( wx_is_at_least_gtk3(22) )
```

Comment:
The old code doesn't use `gdk_monitor_get_scale_factor()`, perhaps there is a corresponding pre-3.22 API for doing the same thing (which one?), but if not, I think it's worth keeping the new version just because of this.